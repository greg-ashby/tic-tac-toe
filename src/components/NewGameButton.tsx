'use client';

type Props = {
  onClick: () => void;
};
export default function NewGameButton({ onClick }: Props) {
  return (
    <button className="text-2xl font-bold" type="button" onClick={onClick}>
      New Game
    </button>
  );
}
