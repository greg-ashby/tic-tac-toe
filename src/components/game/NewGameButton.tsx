type Props = {
  onClick: () => void;
};
export default function NewGameButton({ onClick }: Props) {
  return (
    <button
      className="text-2xl font-bold bg-green-500 text-white px-4 py-2 rounded hover:bg-green-600 transition duration-200"
      type="button"
      onClick={onClick}
    >
      New Game
    </button>
  );
}
