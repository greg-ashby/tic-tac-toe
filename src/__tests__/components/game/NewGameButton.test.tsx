import NewGameButton from '@/components/game/NewGameButton.tsx';
import { fireEvent, render } from '@testing-library/react';

describe('NewGameButton', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<NewGameButton onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('calls onClick when clicked', () => {
    const handleClick = jest.fn();
    const { getByRole } = render(<NewGameButton onClick={handleClick} />);

    const button = getByRole('button', { name: /new game/i });
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
