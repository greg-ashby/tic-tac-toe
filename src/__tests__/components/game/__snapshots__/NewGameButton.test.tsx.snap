// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGameButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="text-2xl font-bold bg-green-500 text-white px-4 py-2 rounded hover:bg-green-600 transition duration-200"
    type="button"
  >
    New Game
  </button>
</DocumentFragment>
`;
